src/utils/mesh_utils.ts

```diff
@@ -69,25 +87,42 @@ export const meshUtils = {
                     break;
                 }
                 case OrderEventEndState.Invalid:
+                    removed.push(apiOrder);
+                    persistentUpdated.push(apiOrder);
+                    break;
                 case OrderEventEndState.Cancelled:
+                    removed.push(apiOrder);
+                    persistentUpdated.push(apiOrder);
+                    break;
                 case OrderEventEndState.Expired:
+                    removed.push(apiOrder);
+                    persistentUpdated.push(apiOrder);
+                    break;
                 case OrderEventEndState.FullyFilled:
+                    removed.push(apiOrder);
+                    persistentUpdated.push(apiOrder);
+                    break;
                 case OrderEventEndState.StoppedWatching:
+                    removed.push(apiOrder);
+                    persistentUpdated.push(apiOrder);
+                    break;
                 case OrderEventEndState.Unfunded: {
                     removed.push(apiOrder);
+                    persistentUpdated.push(apiOrder);
                     break;
```

Comment:
NIT(just my opinon feel free to ignore 😄 ): since we are doing the same thing in all the case statements I would just let it fall through to the bottom case so we don't have to make changes in many different places in the future
```suggestion
                case OrderEventEndState.Invalid:
                case OrderEventEndState.Cancelled:
                case OrderEventEndState.Expired:
                case OrderEventEndState.FullyFilled:
                case OrderEventEndState.StoppedWatching:
                case OrderEventEndState.Unfunded: {
                    removed.push(apiOrder);
                    persistentUpdated.push(apiOrder);
                    break;
```

Reply:
good point! changing